### DISCUSSION TITLE
Is interacting with dynamically added elements supported?

### DISCUSSION THREAD

**Discussion by salarmehr** [php-webdriver/php-webdriver#502] [Q&A] [answered]
I wonder if its possible to for example click on an element that is added to dom by an ajax call? I could not managed to that and I wonder if it's an issue with webdriver its self, i don't create a full issue of it. 

I was assuming it's possible to do any think a real user can do by webdriver. however according to http://mink.behat.org/en/latest/guides/drivers.html#driver-feature-support there are somethings that are not possible to do.

**Comment by OndraM** [comment, marked as answer]
Hi @salarmehr , yes, this is possible.

You must first use [explicit wait](https://github.com/facebook/php-webdriver/wiki/HowTo-Wait), to make sure the item is already present in DOM. Then its no problem interacting with those elements.

BTW Behat uses different Selenium driver than php-webdriver, so its documentation may not be relevant for php-webdriver behavior.

**Comment by salarmehr** [comment]
OMG, I was using `Laravel Dusk`  `waitFor` method assuming it uses php-webdriver explicit waits. However it does not! And for some reasons it sucks in finding loaded element.

Frustrated of using Dusk and wasting time more! :(
